catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrSplitCatalogProvider.java

```diff
@@ -0,0 +1,672 @@
+import ddf.catalog.operation.Request;
+import ddf.catalog.operation.Response;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.CreateResponseImpl;
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import ddf.catalog.operation.impl.SourceResponseImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.operation.impl.UpdateResponseImpl;
+import ddf.catalog.source.CatalogProvider;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceMonitor;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.source.solr.api.SolrCollectionCreationPlugin;
+import ddf.catalog.source.solr.api.SolrCollectionProvider;
+import ddf.catalog.util.impl.MaskableImpl;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.commons.lang.BooleanUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.Validate;
+import org.codice.solr.factory.SolrClientFactory;
+import org.codice.solr.factory.SolrCollectionConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SolrSplitCatalogProvider extends MaskableImpl implements CatalogProvider {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SolrSplitCatalogProvider.class);
+
+  private static final String DESCRIBABLE_PROPERTIES_FILE = "/describable.properties";
+
+  private static final Properties DESCRIBABLE_PROPERTIES = new Properties();
+
+  static {
+    try (InputStream inputStream =
+        SolrSplitCatalogProvider.class.getResourceAsStream(DESCRIBABLE_PROPERTIES_FILE)) {
+      DESCRIBABLE_PROPERTIES.load(inputStream);
+    } catch (IOException e) {
+      LOGGER.info("Did not load properties properly.", e);
+    }
+  }
+
+  protected static final String DEFAULT_COLLECTION = "default";
+
+  protected static final String DEFAULT_QUERY_ALIAS = "catalog";
+
+  protected static final String CATALOG_PREFIX_SEPARATOR = "_";
+
+  @VisibleForTesting static final String GET_HANDLER_WORKAROUND_PROP = "getHandlerWorkaround";
+
+  @VisibleForTesting static final String COLLECTION_ALIAS_PROP = "collectionAlias";
+
+  protected final Map<String, BaseSolrCatalogProvider> catalogProviders = new ConcurrentHashMap<>();
```

Comment:
do we have to worry about concurrency for these other globals?

Reply:
i think we are ok, the other globals are all injected and managed by blueprint